.github/workflows/tag-version.yaml

```diff
@@ -0,0 +1,43 @@
+jobs:
+  tag-version:
+    runs-on: ubuntu-latest
+    if: ${{contains(github.event.head_commit.message, 'Bump setup')}}
+    permissions:
+      id-token: write
+      contents: write
+      pull-requests: write
+      checks: write
+    steps:
+      - id: jwt
+        env:
+          EXPIRATION: 600
+          ISSUER: ${{ secrets.SEMGREP_CI_APP_ID }}
+          PRIVATE_KEY: ${{ secrets.SEMGREP_CI_APP_KEY }}
+        name: Get JWT for semgrep-ci GitHub App
+        uses: docker://public.ecr.aws/y9k7q4m1/devops/cicd:latest
+
+      - id: token
+        name: Get token for semgrep-ci GitHub App
+        run: |
+          TOKEN="$(curl -X POST \
+          -H "Authorization: Bearer ${{ steps.jwt.outputs.jwt }}" \
+          -H "Accept: application/vnd.github.v3+json" \
+          "https://api.github.com/app/installations/${{ secrets.SEMGREP_CI_APP_INSTALLATION_ID }}/access_tokens" | \
+          jq -r .token)"
+          echo "::add-mask::$TOKEN"
+          echo "token=$TOKEN" >> $GITHUB_OUTPUT
+
+      - uses: actions/checkout@v4
+        with:
+          token: ${{ steps.token.outputs.token }}
+
+      - name: Bump version in this repo
+        env:
+          GITHUB_TOKEN: ${{ steps.token.outputs.token }}
+        run: scripts/tag-version.sh "${NEW_SEMGREP_VERSION}"
+
+name: tag-version
+on:
+  push:
+    branches:
+      - develop
```

Comment:
What would happen here if we make multiple commits on a given version?

Reply:
The thought behind this workflow is that it is ran only after the bump version workflow's PR is merged which is just one single squashed merge commit; I'm not sure if there is any more invariants I should be concerned about in this repo:

Reply:
( am still trying to debug this workflow as for some reason it does not trigger after an auto merge, at least thats what I saw in the test-pre-commit repo)

Reply:
Alright I've deleted this workflow: the way this is automated now is

- `bump_version.yml`: Commits on a branch, pushes the branch & tags a commit on the branch
- `autoapprove.yml`: If PR is made by semgrep-ci[bot], approve & wait for checks to pass; then auth as semgrep-ci[bot], move the tag to develop. (delete the tag made on the branch, then push a tag on develop)